src/wp-includes/functions.php

```diff
@@ -4947,14 +4947,31 @@ function _wp_array_get( $input_array, $path, $default_value = null ) {
 	}
 
 	foreach ( $path as $path_element ) {
-		if (
-			! is_array( $input_array ) ||
-			( ! is_string( $path_element ) && ! is_integer( $path_element ) && ! is_null( $path_element ) ) ||
-			! array_key_exists( $path_element, $input_array )
-		) {
+		if ( ! is_array( $input_array ) ) {
 			return $default_value;
 		}
-		$input_array = $input_array[ $path_element ];
+
+		if ( is_string( $path_element )
+			|| is_integer( $path_element )
+			|| null === $path_element
+		) {
+			/*
+			 * Check if the path element exists in the input array.
+			 * We check with `isset()` first as it is a lot faster
+			 * than `array_key_exists()`.
+			 * If `isset()` returns false, we check with `array_key_exists()`
+			 * which also checks for `null` values.
+			 */
```

Comment:
Is the extra `array_key_exists()` check for `null` needed? Because `null` means returning `$default_value` anyway. Or am I missing something?